src/app/components/core/control-dispatcher/add-control-script/add-step/add-configure/add-configure.component.ts

```diff
@@ -200,6 +186,21 @@ export class AddConfigureComponent implements OnInit {
 
   }
 
+  setCategoryName(nodes) {
+    let category = nodes.find(r => r.key == this.config.category);
+    if(!category) {
+      for(let node of nodes){
+        if(node.children.length > 0) {
+          category = this.setCategoryName(node.children);
+          if(category){
```

Comment:
Else is not required here; this is just to check whether the category is found in any of the child node otherwise it will check further in other sibling and child nodes.